requirements.txt

```diff
@@ -0,0 +1 @@
+pre-commit
```

Comment:
Use full version numbers here so we get the same version for each run. If we want to use `requirements.txt` rather then `Pipfile` (pipenv) of `pyproject.toml` (poetry), then we probably should use pip-tools to get the usual un-versioned+lock files combination.

Reply:
added `requirements.lock` with a specific version, changed the workflow to use it instead. 
I tried to use `pip-tools` to compile the requirements file, but currently encountered conflicts from pre-commit's requirements file and `importlib-resources<5.3,>=5.4`.